### ISSUE TITLE
Latest patch 1.1.1 introduces a breaking change and runtime error when running mocha tests

### ISSUE THREAD

**Issue by DimitarTodorov** [glenjamin/mocha-multi#62]
Running mocha tests with latest mocha-multi reporter plugin leads to breaking change and failed run. 
The following error is logged in the console and the run fails.  
```
runner.stats = stats;   
^ TypeError: Cannot set property stats of #<EventEmitter> which has only a getter
```

The error is not thrown in `"mocha-multi": "1.1.0"`.

**Comment by glenjamin** [comment]
This seems to be caused by #61

@fargies - is this working for you?

**Comment by Avyiel** [comment]
I am running into the same error, but opened an issue are the CodeceptJS repo: https://github.com/Codeception/CodeceptJS/issues/1854

Exact same error message, you can see details and example repo in that issue.

**Comment by pmalouin** [comment]
We needed to upgrade mocha from 5.2.0 to 6.2.0 to solve this issue on our build. Seems that `mocha-multi@1.1.1` breaks compatibility with versions before mocha 6.1.4 ([Diff](https://github.com/glenjamin/mocha-multi/compare/v1.1.0...v1.1.1#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R46))

mocha-multi@1.1.0 -> needs mocha < 6.0.0
mocha-multi@1.1.1 -> needs mocha >= 6.1.4

**Comment by Avyiel** [comment]
I spun up a repo to reproduce this issue, currently using mocha 6.2.0: https://github.com/Avyiel/repro-codeceptjs-mocha6/blob/master/yarn.lock#L976

**Comment by Avyiel** [comment]
It appears that someone issued a fix, maybe a PR could happen?

https://github.com/svkarpovich/mocha-multi/commit/485137f8199f27051d0766d59d2dd1392b3e6f86

**Comment by glenjamin** [comment]
Yep, keeping compatibility with older mocha versions was intended.

I’m interested to know what’s causing people to upgrade this module immediately but not mocha itself?

**Comment by mikeharder** [comment]
> I’m interested to know what’s causing people to upgrade this module immediately but not mocha itself?

@glenjamin: I think it's because most people use caret (`^`) ranges for dependencies.  The update to `mocha-multi` was a patch update (`1.1.0` to `1.1.1`), while the update to `mocha` itself was a major update (`5.2.0` to `6.0.0`).  So if a package has these dependencies:

```
"mocha": "^5.2.0",
"mocha-multi": "^1.1.0",
```

It is broken by the update to `mocha-multi`.

**Comment by Avyiel** [comment]
> @glenjamin: I think it's because most people use caret (`^`) ranges for dependencies. The update to `mocha-multi` was a patch update (`1.1.0` to `1.1.1`), while the update to `mocha` itself was a major update (`5.2.0` to `6.0.0`).  [...]

Precisely my situation. 

Would like to add that the [previously linked repository](https://github.com/svkarpovich/mocha-multi/commit/485137f8199f27051d0766d59d2dd1392b3e6f86) solved the issue for me: `"mocha-multi": "svkarpovich/mocha-multi"`

**Comment by glenjamin** [comment]
Published as `v1.1.2`

**Comment by glenjamin** [comment]
Are people using lockfiles, and finding this issue via dependabot or renovate or similar?

**Comment by Avyiel** [comment]
> Are people using lockfiles, and finding this issue via dependabot or renovate or similar?

Yes, we are using a lockfile. Issue popped up in our CI when a pipeline failed the test job, firstly because we don't freeze the lockfile and secondly because of tilde ranges (`~1.1.0` in our case)

**Comment by glenjamin** [comment]
Oh interesting, I'd highly recommend using `yarn install --frozen-lockfile` or `npm ci` on CI environments.

I do appreciate the quick heads up from people finding these issues though!

**Comment by mikeharder** [comment]
@glenjamin: Was the goal of #61 was to add support for `mocha@^6.0.0` without introducing a breaking change?  If so, I believe the [update](https://github.com/glenjamin/mocha-multi/pull/61/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L46-R46) to `peerDependencies` is incorrect, since removing versions of a peer dependency should be considered a breaking change.

```
-    "mocha": ">=2.2.0 <6.0.0"
+    "mocha": "^6.1.4"
```

Instead, I think `peerDependencies` should have been updated as follows, which is not a breaking change:

```
-    "mocha": ">=2.2.0 <6.0.0"
+    "mocha": ">=2.2.0 <7.0.0"
```

PR: https://github.com/glenjamin/mocha-multi/pull/64

**Comment by mikeharder** [comment]
I tried adding tests for `mocha` v2 to v4 in #64, but the tests fail.  Do you think this is product issue, where `mocha-multi` is no longer compatible with these older `mocha` versions?  If so, I think the correct fix is to make a breaking-change update to `peerDependencies` and update the version of `mocha-multi` to `2.0.0`:

```
-    "mocha": ">=2.2.0 <6.0.0"
+    "mocha": "^5.0.0 || ^6.0.0"
```

If you think `mocha-multi` should be compatible with these older `mocha` versions, it might just be that the tests need to be updated accordingly.

Either way, I think the tested versions of `mocha` should match what's listed in `peerDependencies`.

**Comment by glenjamin** [comment]
Yeah, I generally avoid breaking back compat unless there is a strong need to.

I also generally prefer to leave peerDependencies open

I admit I didn’t look over #61 as closely as intended, I was wary of the extra changes but should have tested in a bit more detail.

I’ll take a look at your PR when I get a mo

**Comment by mikeharder** [comment]
I think the failures on older versions of `mocha` are just an issue with the test script, fixed here:

https://github.com/glenjamin/mocha-multi/pull/64/commits/a3cf7f8015c7e7d57498561c44e4f0b0211ca4de

**Comment by mikeharder** [comment]
@glenjamin: Thanks for merging my PR.  When do you expect to publish the next version?

**Comment by glenjamin** [comment]
Published as `1.1.3` 👍

**Comment by fargies** [comment]
> This seems to be caused by #61
> 
> @fargies - is this working for you?

I didn't saw that one, thanks for the fix
